monitoring/finished_build_stats_publisher.py

```diff
@@ -0,0 +1,24 @@
+import time
+
+
+class MaxRetriesExceededError(Exception):
+    pass
+
+
+class FinishedBuildStatsPublisher(object):
+
+    def __init__(self, collector, publisher):
+        self.collector = collector
+        self.publisher = publisher
+
+    def collect_and_publish(self, delay_ms=1000, max_retries=10):
+        while max_retries != 0:
+            stats = self.collector.collect()
+            if stats.is_finished():
```

Comment:
Also nitpicky, but maybe it should be .is_build_finished? Since stats can't be finished.